online_party_system/party_match.go

```diff
@@ -452,3 +635,21 @@ func (p PartyMatch) MatchTerminate(ctx context.Context, logger runtime.Logger, d
 	}
 	return nil
 }
+
+func SendResponse(partyMsg PartyMatchMessage, responseCode int64, to []runtime.Presence, logger runtime.Logger, dispatcher runtime.MatchDispatcher) {
+	// Only send a response if party message ID is set
+	if partyMsg.Id > 0 {
```

Comment:
This is to pair up "requests" with "responses" in the client, if I understand correctly. Under what scenarios will the `Id` not be set?

Reply:
I set this up so you can opt-in to the callbacks per message. That way the client can decide which callbacks it requires and nakama can skip the marshaling and broadcasting of messages where a callback is unnecessary. Also a bit of backward compatibility to support the old message format of []byte which had no ID sent

Reply:
@zkennyanderson I've thought it through and discussed it with the team and we feel it's best if we drop the ability to omit response messages if no `Id` is set, even if other consumers of this package will need to do some upgrade work client-side. Having all input be a `PartyMatchMessage` type feels like the most consistent path forward. Clients that don't care about the response can just ignore it when it's returned.

Reply:
Thinking about it, this makes sense although it leaves me with some typing to do.